pkg/amdsevsnp/report.go

```diff
@@ -176,9 +212,50 @@ func FetchRawSNPReport(reportData []byte) ([]byte, error) {
 		Error:           0,
 	}
 
-	if err := linux.Ioctl(f, reportCode|ioctlBase, unsafe.Pointer(payload)); err != nil {
+	if err := linux.Ioctl(f, snpGetReportIoctlCode5, unsafe.Pointer(payload)); err != nil {
+		return nil, err
+	}
+	return msgReportOut.Report[:], nil
+}
+
+func fetchRawSNPReport6(reportData []byte) ([]byte, error) {
+	f, err := os.OpenFile(snpDevicePath6, os.O_RDWR, 0)
+	if err != nil {
+		return nil, err
+	}
+	defer func() {
+		_ = f.Close()
+	}()
+
+	var (
+		msgReportIn  reportRequest
+		msgReportOut reportResponse
+	)
+	reportOutContainer := [reportResponseContainerLength6]byte{}
+
+	if reportData != nil {
+		if len(reportData) > len(msgReportIn.ReportData) {
+			return nil, fmt.Errorf("reportData too large: %s", reportData)
+		}
+		copy(msgReportIn.ReportData[:], reportData)
+	}
+
+	payload := &guestRequest6{
+		MsgVersion:   1,
+		RequestData:  unsafe.Pointer(&msgReportIn),
+		ResponseData: unsafe.Pointer(&reportOutContainer),
+		Error:        0,
+	}
+
+	if err := linux.Ioctl(f, snpGetReportIoctlCode6, unsafe.Pointer(payload)); err != nil {
 		return nil, err
 	}
+
+	reportOutSize := unsafe.Sizeof(msgReportOut)
+	reportOutPtr := unsafe.Pointer(&msgReportOut)
+	reportOutAsSlice := (*[reportResponseContainerLength6]byte)(reportOutPtr)
+	copy(reportOutAsSlice[:reportOutSize], reportOutContainer[:reportOutSize])
```

Comment:
this will copy the result, and handle all the size computations:

```suggestion
    msgReportOut := *(*reportResponse)(unsafe.Pointer(&reportOutContainer[0])
```

Reply:
I don't think this matters at all and is not worth changing prior to merging.
This PR is a blocker for the 6.1 support which we REALLY need to get in.